c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -482,6 +482,32 @@ class CachingAllocatorConfig {
   }
 };
 
+// records allocate/free events
+class MemoryEventTracker {
+ private:
+  std::mutex mutex;
+
+ public:
+  std::vector<std::vector<AllocFreeEvent>> alloc_free_events;
+  void append_alloc_free_event(
+      intptr_t ptr,
+      size_t size,
+      bool type,
+      int device) {
+    std::lock_guard<std::mutex> lock(mutex);
+    alloc_free_events[device].push_back(AllocFreeEvent{
```

Comment:
since we already have device in the event struct, do we still need to make `alloc_free_events` use device as the 1st layer of index?

Reply:
I removed the device from the event struct. I think it is more convenient to separate events for different devices? Please let me know if you prefer to use one layer for the list of events and include the device in the event struct. Thank you.

Reply:
I don't have strong opinion on this one. Please feel free to use the one you prefer.

Reply:
Thank you.